File: go/MajorityElement/MajorityElement.go
package main

import "fmt"

// Time -> 0(N) Space -> 0(N)
func majorityElement(nums []int) int {
	counter := make(map[int]int, 0)

	for _, num := range nums {
		counter[num]++
		if counter[num] > len(nums)/2 {
			return num
		}
	}
	return 0
}

func main() {
	fmt.Println(majorityElement([]int{1, 2, 2}))
}

File: python/majorityElement.py
# Time -> 0(N) Space -> 0(N)
def majorityElement(nums: list[int]) -> int:
    counter = {}
    for num in nums:
        counter[num] = counter.get(num, 0) + 1
        if counter[num] > (len(nums) // 2):
            return num
    return 0


print(majorityElement([1, 2, 2, 2]))
